File: GotDotNet.com/UrlBuddy/C#/UrlMenuItem.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;

namespace UrlBuddy
{
	public class UrlMenuItem : ToolStripMenuItem
	{
		private UrlInfo _urlInfo;

		public UrlMenuItem(UrlInfo urlInfo) : base(urlInfo.Title)
		{
			_urlInfo = urlInfo;
		}

		protected override void OnClick(EventArgs e)
		{
			Process.Start(_urlInfo.Url);
		}

		public UrlInfo UrlInfo
		{
			get { return _urlInfo; }
		}
	}
}

File: GotDotNet.com/UrlBuddy/VB.NET/UrlMenuItem.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms
Imports System.Diagnostics

Public Class UrlMenuItem
    Inherits ToolStripMenuItem

    Private _urlInfo As UrlInfo
    
    Public Sub New(urlInfo As UrlInfo)
        MyBase.New(urlInfo.Title)
        _urlInfo = urlInfo
    End Sub

    Protected Overrides Sub OnClick(e As EventArgs)
        Process.Start(_urlInfo.Url)
    End Sub
    
    Public ReadOnly Property UrlInfo() As UrlInfo
        Get
            Return _urlInfo
        End Get
    End Property

End Class